pages/src/App.js

```diff
@@ -11,39 +11,54 @@ class App extends Component {
   }
 
   componentWillMount () {
-    fetch('/todos').then(res =>
-      res.json().then(todos => {
-        this.setState({
-          todos: new Map().withMutations(map => {
+
+    // Create WebSocket connection.
+    let socket = new WebSocket('ws://localhost:3000');
+    this.socket = socket
+
+    // Receiving a message from the server
+    socket.onmessage = msg => {
+      let todos = JSON.parse(msg.data);
+      let {id} = todos
+      console.log(todos);
+      this.setState({ todos: Map(todos)})
+      /*
+      if(Object.keys(todos).length>0){
+      this.setState({ todos: new Map().withMutations(map => {
             todos.forEach(todo => {
               map.set(todo.id, todo)
             })
           })
-        })
-      })
-    )
-  }
+        })}*/
+      console.log("todos");
+    }
 
-  _onCreateTodo = todo => {
-    const { id } = todo
-    const { todos } = this.state
-    const previous = todos.get(id)
+    // Connection opened
+    socket.onopen = () => {
+      let msg = {data : "send me data"}
+      this.socket.send(JSON.stringify(msg));
+    }
+  }
 
-    this.setState({ todos: todos.set(id, todo) })
 
-    fetch(`/todos/${id}`, {
-      body: JSON.stringify(todo),
-      headers: {
-        'content-type': 'application/json'
-      },
-      method: 'PUT'
-    }).catch(() => {
-      this.setState({
-        todos: previous === undefined
-          ? todos.delete(id)
-          : todos.set(id, previous)
-      })
+  _onTodoRemove = id => {
+    //this.state = todos.filter((todo) => {})
+    // filter out the todo that has to be removed
+    let list = new Map()
```

Comment:
This line is completely unnecessary because you are never using the just constructed `Map`.